reactor-core/src/main/java/reactor/core/publisher/Mono.java

```diff
@@ -912,6 +913,28 @@ public static <T, D> Mono<T> using(Callable<? extends D> resourceSupplier,
 		return using(resourceSupplier, sourceSupplier, resourceCleanup, true);
 	}
 
+	/**
+	 * Uses a {@link AutoCloseable} resource, generated by a supplier for each individual Subscriber,
+	 * while streaming the value from a Mono derived from the same resource and makes sure
+	 * the resource is released if the sequence terminates or the Subscriber cancels.
+	 * <p>
+	 * Unlike in {@link Flux#using(Callable, Function, Consumer) Flux}, in the case of a valued {@link Mono} the cleanup
+	 * happens just before passing the value to downstream. In all cases, exceptions raised by the cleanup
+	 * {@link Consumer} may override the terminal event, discarding the element if the derived {@link Mono} was valued.
+	 * <p>
+	 * <img class="marble" src="doc-files/marbles/usingForMono.svg" alt="">
+	 *
+	 * @param resourceSupplier a {@link Callable} that is called on subscribe to create the resource
+	 * @param sourceSupplier a {@link Mono} factory to create the Mono depending on the created resource
+	 * @param <T> emitted type
+	 * @param <D> resource type
+	 *
+	 * @return new {@link Mono}
+	 */
+	public static <T, D extends AutoCloseable> Mono<T> using(Callable<D> resourceSupplier,
```

Comment:
```suggestion
	public static <T, D extends AutoCloseable> Mono<T> using(Callable<? extends D> resourceSupplier,
```

Reply:
I thik it would be better with a  `R` than `D`.

Reply:
https://github.com/reactor/reactor-core/blob/70f512672cfc2e0f5ba1ddce1f3ee25b3a9e7f9b/reactor-core/src/main/java/reactor/core/publisher/Flux.java#L2098

`<T, D> Flux<T> using` already uses `D` and I just copy it 😅
If we decide to use `R instead D`, we need to fix other methods..

Reply:
Agreed, I believe consistency is preferred.